django/thunderstore/repository/api/experimental/views.py

```diff
@@ -60,3 +68,31 @@ def get_no_cache_response(cls):
 
     def get_queryset(self):
         return get_mod_list_queryset(self.request.community_site)
+
+
+class UploadPackageApiView(APIView):
+    """
+    Uploads a package. Requires multipart/form-data.
+    """
+
+    parser_classes = [MultiPartParser]
+
+    def post(self, request):
+        categories = request.data.get("categories")
+        if categories:
+            try:
+                request.data["categories"] = json.loads(categories)
+            except json.JSONDecodeError:
+                return Response(
+                    {"error": "Invalid"}, status=status.HTTP_400_BAD_REQUEST
+                )
```

Comment:
This is probably inconsistent with other errors

Reply:
Looks like it